teamworkApp/lib/importData.py

```diff
@@ -1,26 +1,234 @@
+	This processing requires very specifically formatted .csv files and it
+	assumes they come from the results of the Teamwork Survey sent out by
+	muddersOnRails.
+	input csvFilename: name (and possibly path to) the csv file corresponding
+		to the data received by respondants to the Teamwork Survey
+	output studentDict: a dictionary. The keys are usernames of students
+		collected from the Teamwork Survey. The value for each key is a list of
+		three pieces of information:
+			[timestamp, expandedResponses, enummedResponses]
+			timestamp: the time at which the student filled out the Teamwork
+				Survey.
+			expandedResponses: the responses to each question expanded out into
+				a string of the four-digit integer corresponding to their
+				rankings, this is a list of 18 responses.
+			enummedResponses: the list of responses to each question enumerated
+				according to the IntEnum Answer_Value. This is a list of 18
+				integers corresponding to the IntEnum."""
+	with open(csvFileName) as csvfile:
+
+		timestamps        = []
+		expandedResponses = []
+		studentDict       = {}
+
+		answerReader = csv.reader(csvfile)
+		studentData  = list(answerReader)
+		# get rid of the row with questions
+		studentData  = studentData[1:]
+
+		for student in studentData:
+			# these are the 18x4 responses to the survey
+			teamworkResponses = student[1:73]
+			expandedResponses = []
+			enummedResponses  = []
+			count = 0
+			while (count < 72):
+				expandedResponses += [ teamworkResponses[count]   +
+								       teamworkResponses[count+1] +
+								       teamworkResponses[count+2] +
+								       teamworkResponses[count+3] ]
+				count += 4
+			for resp in expandedResponses:
+				ans = getAnswer(resp)
+				enummedResponses += [ans]
 
+			# the timestamps are saved in the first column of the .csv file
+			timestamp = student[0]
+			studentDict[student[-1]] = [timestamp, expandedResponses, enummedResponses]
+
+		return studentDict
+
+def findStyle(expandedResponses):
+	contributor  = 0
+	collaborator = 0
+	communicator = 0
+	challenger   = 0
+	tracker      = 0
+	for resp in expandedResponses:
+		# this pattern is based on the teamwork survey scoring scheme
+		if tracker == 0:
+			contributor  += int(resp[0]) # a
+			collaborator += int(resp[1]) # b
+			communicator += int(resp[2]) # c
+			challenger   += int(resp[3]) # d
+		elif tracker == 1:
+			contributor  += int(resp[3]) # d
+			collaborator += int(resp[0]) # a
+			communicator += int(resp[1]) # b
+			challenger   += int(resp[2]) # c
+		elif tracker == 2:
+			contributor  += int(resp[2]) # c
+			collaborator += int(resp[3]) # d
+			communicator += int(resp[0]) # a
+			challenger   += int(resp[1]) # b
+		elif tracker == 3:
+			contributor  += int(resp[1]) # b
+			collaborator += int(resp[2]) # c
+			communicator += int(resp[3]) # d
+			challenger   += int(resp[0]) # a
+		else:
+			tracker = -1
+		tracker += 1
+	return communicator, collaborator, challenger, contributor
```

Comment:
I think that I implemented most of this in evaluateAnswer.py. I think that rather than reimplementing it here, we should choose one way of doing it (I used mods and other python structures to decrease the number of lines) and use it in both places. This might require some refactoring, but I think it will be better in the long run.

Reply:
Ok, I change this to import your file and call your functions.

Reply:
Has evaluateResponses been updated since we updated the schema? I'm using the develop branch version so I'm thinking it hasn't.

I've added the code and line that would call your `find_styles`, but it doesn't return the proper information at the moment so I've got it commented out. I didn't want to refactor your code and I acknowledged that while my code works, it is essentially using deprecated code by calling my version of the function. So once we get that sorted out we should be good!

I've also realized that I can't really understand the details and inner-workings of evalutateAnswers because it is under-documented. So when you (or whoever) goes back to refactor it keep that in mind.

Reply:
I dont think its been updated recently, sorry about the commenting. I dont have time to fix it this week but I could go through next week and help with documentation for current code. If you want to leave in your code for now thats fine, but maybe add a todo or task on KanbanFlow so that we remember to change it.